daemon/peer.c

```diff
@@ -341,13 +340,16 @@ static void peer_breakdown(struct peer *peer)
 	
 	/* If we have a closing tx, use it. */
 	if (peer->closing.their_sig) {
+		const struct bitcoin_tx *close = mk_bitcoin_close(peer, peer);
 		log_unusual(peer->log, "Peer breakdown: sending close tx");
-		broadcast_tx(peer, bitcoin_close(peer), NULL);
+		broadcast_tx(peer, close, NULL);
+		tal_free(close);
```

Comment:
Actually, that was changed in 23107b7cdac5a57d265e8904bcdccdf4c6fbf4cc and this is just closing those leaks.  I will move the comment change to that commit, and document it better so this patch makes sense...